gapic/src/main/com/google/gapid/views/PerformanceView.java

```diff
@@ -547,4 +598,167 @@ protected void okPressed() {
+        ctx.trace("Counter", () -> {
+          int selectedMetricId = selectedMetric == null ? -1 : selectedMetric.getId();
+          if (selectedEntry == null
+              || !selectedEntry.getMetricToValueMap().containsKey(selectedMetricId)
+              || !counterDataLookup.containsKey(selectedMetricId)) {
+            return;
+          }
+          Service.ProfilingData.Counter counter = counterDataLookup.get(selectedMetricId);
+          Service.ProfilingData.GpuCounters.Perf perf = selectedEntry.getMetricToValueMap().get(selectedMetricId);
+          // Use TreeSet so that the indexes are sorted during iteration.
+          TreeSet<Integer> indexes = Sets.newTreeSet(perf.getEstimateSamplesMap().keySet());
+          if (indexes.size() == 0) {
+            return;
+          }
+          double max = indexes.stream().mapToDouble(counter::getValues).max().getAsDouble();
+
+          double h = height - 2 * TRACK_HEIGHT_MARGIN - TRACK_X_AXIS_HEIGHT;
+          double w = width - 2 * TRACK_WIDTH_MARGIN - TRACK_Y_AXIS_WIDTH;
+          double top = TRACK_HEIGHT_MARGIN, bottom = height - TRACK_X_AXIS_HEIGHT - TRACK_HEIGHT_MARGIN;
+          double left = TRACK_Y_AXIS_WIDTH + TRACK_WIDTH_MARGIN, right = width - TRACK_WIDTH_MARGIN;
+          double stepX = w / (indexes.size());
+          double stepY = h / AXIS_Y_CHUNK_NUM;
+
+          // Draw graph.
+          mainGradient().applyBaseAndBorder(ctx);
+          ctx.path(path -> {
+            double lastX = left;
+            path.moveTo(left, bottom);
+            for (int i : indexes) {
+              double nextX = lastX + stepX;
+              double nextY = top + h * (1 - counter.getValues(i) / max);
+              path.lineTo(lastX, nextY);  // Go up.
+              path.lineTo(nextX, nextY);  // Go right.
+              ctx.drawTextCenteredRightTruncate(Fonts.Style.Normal,
+                  "(" + perf.getEstimateSamplesMap().get(i) + ")",
+                  lastX, nextY - 20, stepX, 20);
+              lastX = nextX;
+            }
+            path.lineTo(lastX, bottom);
+            ctx.fillPath(path);
+            ctx.drawPath(path);
+          });
+
+          // Draw x-axis.
+          threadStateSleeping().applyBaseAndBorder(ctx);
+          ctx.drawLine(left, bottom, right + AXIS_ARROW_LENGTH, bottom);
+          ctx.drawLine(right + AXIS_ARROW_LENGTH, bottom, right + AXIS_ARROW_LENGTH - AXIS_ARROW_SIZE, bottom + AXIS_ARROW_SIZE);
+          ctx.drawLine(right + AXIS_ARROW_LENGTH, bottom, right + AXIS_ARROW_LENGTH - AXIS_ARROW_SIZE, bottom - AXIS_ARROW_SIZE);
+          double x = left;
+          for (int i : indexes) {
+            // A counter sample at time {tn} is seen as holding the value {vn} during the implicit
+            // time range between {tn-1} ~ {tn}.
+            if (i > 0) {
+              ctx.drawText(Fonts.Style.Normal, Unit.NANO_SECOND.format(
+                  counter.getTimestamps(i - 1) - traceStart), x, bottom + AXIS_ARROW_SIZE);
+            }
```

Comment:
Looks like `counter.getTimestamps(i - 1) - traceStart` may be negative, is that expected?

![perftab-negative-start](https://user-images.githubusercontent.com/6726747/107955221-72b1d780-6f95-11eb-9712-cc49021956b3.png)

Reply:
Yes, this is expected. ;) the time start based on the start time of the first gpu slice, but the counter samples covers longer time. this exactly counter sample covers partly with the first gpu slice so I still count its contribution. 
But considering that this may cause confusion, I may need to consult Jay for better practices on UX, also for the above explanation issue, and address them later in future PR.

Reply:
OK to fix/polish that later.